randovania/layout/preset_migration.py

```diff
@@ -840,6 +840,12 @@ def _fix(target):
 def _migrate_v53(preset: dict) -> dict:
     return _update_default_dock_rando_for_game(preset, RandovaniaGame.METROID_PRIME_ECHOES)
 
+def _migrate_v54(preset: dict) -> dict:
+    game = preset["game"]
+    if game in {"blank", "cave_story", "am2r"}:
+        return preset
+    preset["configuration"]["dock_rando"]["types_state"]["teleporter"] = {"can_change_from": [],"can_change_to": []}
+    return preset
```

Comment:
instead, you should update the `default_dock_lock_settings` in `migration_data` for each game and just `return _update_default_dock_rando(preset)`

Reply:
Hmmm that does not work for me. `_update_default_dock_rando` is calling `_update_default_dock_rando_for_game` which has the mode hardcoded to `vanilla`.
E.g. `test\games\dread\exporter\test_dread_patch_data_factory.py` has the test `test_create_patch_with_crazy_settings` and the preset gets changed by that to vanilla mode and test fails.

AFAIK that also would reset the settings (like `can_change_from`) in the preset if we just change it back to the default from the `migration_data`.

Reply:
ah, good point about resetting presets. perhaps a function that updates the defaults for a particular dock type would be warranted? idk. if we ship it like this i can live with it

Reply:
I don't like this reusing of `_update_default_dock_rando_for_game` every time as it causes old migrations to have different results every time we update migration_data, and that causes what is happening in migrations harder to reason about.

Reply:
Then I will leave it as it is?
I have to remove the `teleporter` part from the migration_data files again as it was only to test the `_update_default_dock_rando_for_game`